### PR TITLE
Fix custom umbraco dictionary keys not working once again

### PR DESCRIPTION
**Pull request by GianniDPC**
I tested with your latest changes and it stopped working again.

I think it is because of the following (Required attribute as example):

Below code will execute only once and for some reason the ErrorMessage will always be the default one even when the DictionaryKey property has been set on my model.
```
 public UmbracoRequiredAttribute() 
		: base()
	{
		ErrorMessage = UmbracoDictionary.GetDictionaryValue(DictionaryKey);
	}
```
If we put that line over here, it will execute everytime the page is reloaded. This seems to work just fine and reads the value set in the dictionary. I tested it and it does not seem to break the default behaviour if no DictionaryKey has been set on the model.
```
public IEnumerable<ModelClientValidationRule> GetClientValidationRules(ModelMetadata metadata, ControllerContext context)
{
	ErrorMessage = UmbracoDictionary.GetDictionaryValue(DictionaryKey);
	yield return new ModelClientValidationRequiredRule(FormatErrorMessage(metadata.GetDisplayName()));
}
```

So in conclusion while I was debugging and once the UmbracoRequiredAttribute constructor was executed the dictionary key was still the default one which is "RequiredError" in this case, even though I set the property on my model to be something else.

It only seems to be applying the value I set on the property in the GetClientValidationRules method. (for some unknown reason, because I also don't understand why it shouldn't be able to read my set property value when executing the constructor.)

### CODE REVIEW

**Comment by rasmuseeg** [comment]
Hi thanks for reporting though I do wonder why the issue appears now, and what version of Umbraco you are using?
________________________________

**Comment by GianniDPC** [comment]
> Hi thanks for reporting though I do wonder why the issue appears now, and what version of Umbraco you are using?
> […](#)
> ________________________________ From: GianniDPC <notifications@github.com> Sent: Sunday, October 13, 2019 12:55:37 PM To: rasmuseeg/Our.Umbraco.DataAnnotations <Our.Umbraco.DataAnnotations@noreply.github.com> Cc: Subscribed <subscribed@noreply.github.com> Subject: [rasmuseeg/Our.Umbraco.DataAnnotations] Fix custom umbraco dictionary keys not working once again (#9) I tested with your latest changes and it stopped working again. I think it is because of the following (Required attribute as example): Below code will execute only once and for some reason the ErrorMessage will always be the default one even when the DictionaryKey property has been set on my model. public UmbracoRequiredAttribute() : base() { ErrorMessage = UmbracoDictionary.GetDictionaryValue(DictionaryKey); } If we put that line over here, it will execute everytime the page is reloaded. This seems to work just fine and reads the value set in the dictionary. I tested it and it does not seem to break the default behaviour if no DictionaryKey has been set on the model. public IEnumerable<ModelClientValidationRule> GetClientValidationRules(ModelMetadata metadata, ControllerContext context) { ErrorMessage = UmbracoDictionary.GetDictionaryValue(DictionaryKey); yield return new ModelClientValidationRequiredRule(FormatErrorMessage(metadata.GetDisplayName())); }
> ________________________________ You can view, comment on, or merge this pull request online at: #9<https://nam10.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Frasmuseeg%2FOur.Umbraco.DataAnnotations%2Fpull%2F9&data=02%7C01%7C%7C729c792300cd4ce5b38a08d74fcbe1aa%7C84df9e7fe9f640afb435aaaaaaaaaaaa%7C1%7C0%7C637065609387303101&sdata=bMjDqFN9%2FqDWa7LpZ32x3oh9Xvossd1fvEsVousfh1A%3D&reserved=0> Commit Summary * Fix custom umbraco dictionary keys not working once again File Changes * M src/Our.Umbraco.DataAnnotations/UmbracoCompareAttribute.cs<https://nam10.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Frasmuseeg%2FOur.Umbraco.DataAnnotations%2Fpull%2F9%2Ffiles%23diff-0&data=02%7C01%7C%7C729c792300cd4ce5b38a08d74fcbe1aa%7C84df9e7fe9f640afb435aaaaaaaaaaaa%7C1%7C0%7C637065609387303101&sdata=HxA15BF5P22gCC0qWlmRt%2BRfV03YsiUhv%2FqImKN3WCw%3D&reserved=0> (3) * M src/Our.Umbraco.DataAnnotations/UmbracoMaxLengthAttribute.cs<https://nam10.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Frasmuseeg%2FOur.Umbraco.DataAnnotations%2Fpull%2F9%2Ffiles%23diff-1&data=02%7C01%7C%7C729c792300cd4ce5b38a08d74fcbe1aa%7C84df9e7fe9f640afb435aaaaaaaaaaaa%7C1%7C0%7C637065609387313110&sdata=Endlq8Go4Po6EEl%2FC%2FrQ11N2GBTqoSlYzvZZUgOFCo0%3D&reserved=0> (3) * M src/Our.Umbraco.DataAnnotations/UmbracoMinLengthAttribute.cs<https://nam10.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Frasmuseeg%2FOur.Umbraco.DataAnnotations%2Fpull%2F9%2Ffiles%23diff-2&data=02%7C01%7C%7C729c792300cd4ce5b38a08d74fcbe1aa%7C84df9e7fe9f640afb435aaaaaaaaaaaa%7C1%7C0%7C637065609387323116&sdata=Fp2q%2BM%2BNGu7pTQ7WRGMRUyMNwPSkzI2NuOkAdKuuPcM%3D&reserved=0> (3) * M src/Our.Umbraco.DataAnnotations/UmbracoRangeAttribute.cs<https://nam10.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Frasmuseeg%2FOur.Umbraco.DataAnnotations%2Fpull%2F9%2Ffiles%23diff-3&data=02%7C01%7C%7C729c792300cd4ce5b38a08d74fcbe1aa%7C84df9e7fe9f640afb435aaaaaaaaaaaa%7C1%7C0%7C637065609387323116&sdata=f7JAb2zuFZ2%2BeVHUGgwMBIyo08f9Df5tNswttxUcJ00%3D&reserved=0> (3) * M src/Our.Umbraco.DataAnnotations/UmbracoRegularExpressionAttribute.cs<https://nam10.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Frasmuseeg%2FOur.Umbraco.DataAnnotations%2Fpull%2F9%2Ffiles%23diff-4&data=02%7C01%7C%7C729c792300cd4ce5b38a08d74fcbe1aa%7C84df9e7fe9f640afb435aaaaaaaaaaaa%7C1%7C0%7C637065609387333125&sdata=NK9Qws7riFN5j1%2F3aR5Rr3jD7O0FAU06yspo8Vkb1wE%3D&reserved=0> (3) * M src/Our.Umbraco.DataAnnotations/UmbracoRequiredAttribute.cs<https://nam10.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Frasmuseeg%2FOur.Umbraco.DataAnnotations%2Fpull%2F9%2Ffiles%23diff-5&data=02%7C01%7C%7C729c792300cd4ce5b38a08d74fcbe1aa%7C84df9e7fe9f640afb435aaaaaaaaaaaa%7C1%7C0%7C637065609387333125&sdata=UdD5IL4pjb%2Fivc7MXb%2B6%2FaO%2BA3jrXKCn%2BosnCIfKHLw%3D&reserved=0> (2) * M src/Our.Umbraco.DataAnnotations/UmbracoStringLengthAttribute.cs<https://nam10.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Frasmuseeg%2FOur.Umbraco.DataAnnotations%2Fpull%2F9%2Ffiles%23diff-6&data=02%7C01%7C%7C729c792300cd4ce5b38a08d74fcbe1aa%7C84df9e7fe9f640afb435aaaaaaaaaaaa%7C1%7C0%7C637065609387343193&sdata=kIIpK798jceAygnW%2BQdJNwV%2Fi9mPL%2FPHpKAtS9BZ%2FYU%3D&reserved=0> (3) Patch Links: * #9.patch<https://nam10.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Frasmuseeg%2FOur.Umbraco.DataAnnotations%2Fpull%2F9.patch&data=02%7C01%7C%7C729c792300cd4ce5b38a08d74fcbe1aa%7C84df9e7fe9f640afb435aaaaaaaaaaaa%7C1%7C0%7C637065609387343193&sdata=ZjwaFoSQtaQ2nP4EXNXN6T%2Bx9ldC5yTmOOWgQ7pDtuw%3D&reserved=0> * #9.diff<https://nam10.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Frasmuseeg%2FOur.Umbraco.DataAnnotations%2Fpull%2F9.diff&data=02%7C01%7C%7C729c792300cd4ce5b38a08d74fcbe1aa%7C84df9e7fe9f640afb435aaaaaaaaaaaa%7C1%7C0%7C637065609387353136&sdata=hqWKm8aqdgq%2F0TLVxrWisiYKQBMXfVM%2B8y43v5nha2I%3D&reserved=0> — You are receiving this because you are subscribed to this thread. Reply to this email directly, view it on GitHub<https://nam10.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Frasmuseeg%2FOur.Umbraco.DataAnnotations%2Fpull%2F9%3Femail_source%3Dnotifications%26email_token%3DAALQ6KDNNZQGZCNXQXXDBKDQOL5CTA5CNFSM4JAGO662YY3PNVWWK3TUL52HS4DFUVEXG43VMWVGG33NNVSW45C7NFSM4HRNUQKQ&data=02%7C01%7C%7C729c792300cd4ce5b38a08d74fcbe1aa%7C84df9e7fe9f640afb435aaaaaaaaaaaa%7C1%7C0%7C637065609387353136&sdata=0P0iWbalAn3ZdczI5qh8PCeiFHwKj8JYtG9zESvM404%3D&reserved=0>, or unsubscribe<https://nam10.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fnotifications%2Funsubscribe-auth%2FAALQ6KFTQNSSUGKJ5MOBNALQOL5CTANCNFSM4JAGO66Q&data=02%7C01%7C%7C729c792300cd4ce5b38a08d74fcbe1aa%7C84df9e7fe9f640afb435aaaaaaaaaaaa%7C1%7C0%7C637065609387363145&sdata=GbkFckh9D2NbCJMaQWTCDDp2q88euJLHQHHqqu7FKNc%3D&reserved=0>.

Still Umbraco version 7.13.2.

**Comment by rasmuseeg** [comment]
Okay, this is great. I'll merge
